### PR TITLE
add frictionless support 

### PR DESCRIPTION
**Pull request by emmadickson**
supports https://github.com/webrecorder/wacz-format/issues/20

### CODE REVIEW

File: py-wacz/tests/test_wacz.py [hunk 1]
```diff
@@ -27,44 +16,32 @@ def setUpClass(self, mock_now):
...
             zip_ref.close()
-        
+
         self.wacz_file = os.path.join(TEST_DIR, 'example.wacz')
         self.warc_file = os.path.join(TEST_DIR, 'example-collection.warc')
-        self.wacz_archive = os.path.join(os.path.dirname(os.path.realpath(__file__)), "fixtures/unzipped_wacz/archive/example-collection.warc")
-        self.wacz_index_cdx = os.path.join(os.path.dirname(os.path.realpath(__file__)), "fixtures/unzipped_wacz/indexes/index.cdx.gz")
-        self.wacz_index_idx = os.path.join(os.path.dirname(os.path.realpath(__file__)), "fixtures/unzipped_wacz/indexes/index.idx")
-        self.wacz_yaml = os.path.join(os.path.dirname(os.path.realpath(__file__)), "fixtures/unzipped_wacz/webarchive.yaml")
+        self.wacz_archive = os.path.join(os.path.dirname(os.path.realpath(__file__)), "fixtures/unzipped_wacz/data/archive/example-collection.warc")
```
Comment (ikreymer) [inline py-wacz/tests/test_wacz.py:22]: Frictionless requires all data to be in `data` subdirectory, right?

File: py-wacz/wacz/util.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import hashlib 
+
+def support_hash_file(data):
+    '''Hashes the passed content using sha224'''
+    hashlib.sha224(data).hexdigest()
```
Comment (ikreymer) [inline py-wacz/wacz/util.py:5]: Just wondering, do we want to standardize on sha-224?
I've been generating sha-256 in other places (warcio.js), maybe should just use sha-256? Or provide options?

File: py-wacz/tests/test_wacz.py (hunk 1 above)
Comment (emmadickson) [reply py-wacz/tests/test_wacz.py:22]: yup!

File: py-wacz/wacz/util.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import hashlib 
+
+def support_hash_file(data):
+    '''Hashes the passed content using sha256'''
+    return hashlib.sha256(data).hexdigest()
```
Comment (emmadickson) [inline py-wacz/wacz/util.py:5]: @ikreymer I updated it to sha256. I think it's better to stay consistent.

**Comment by emmadickson** [comment]
Need to note these changes in the documentation before its merged

File: py-wacz/wacz/waczindexer.py [hunk 4]
```diff
@@ -196,25 +195,39 @@ def serialize_json_pages(self, pages):
...
+        package_dict['resources'] = []
+        for i in range(0, len(wacz.infolist())):
+            file = wacz.infolist()[i]
+            package_dict['resources'].append({})
+            package_dict['resources'][i]['path'] = file.filename
+            with wacz.open(file, 'r') as myfile:
+                package_dict['resources'][i]['stats'] = {}
+                package_dict['resources'][i]['stats']['hash'] = support_hash_file(myfile.read())
+                package_dict['resources'][i]['stats']['bytes'] = len(myfile.read())
+                package_dict['resources'][i]['hashing'] = 'sha224'
```
Comment (ikreymer) [inline py-wacz/wacz/waczindexer.py:211]: should be sha256 here

**Comment by ikreymer** [comment]
Noticed one issue, probably a typo, running with the `-t` (extract text) option results in an error:
```
  File "/Users/ilya/venvs/py37-2/bin/wacz", line 11, in <module>
    load_entry_point('wacz==0.1.0', 'console_scripts', 'wacz')()
  File "/Users/ilya/venvs/py37-2/lib/python3.6/site-packages/wacz-0.1.0-py3.6.egg/wacz/main.py", line 28, in main
  File "/Users/ilya/venvs/py37-2/lib/python3.6/site-packages/wacz-0.1.0-py3.6.egg/wacz/main.py", line 87, in create_wacz
  File "/Users/ilya/venvs/py37-2/lib/python3.6/site-packages/wacz-0.1.0-py3.6.egg/wacz/waczindexer.py", line 224, in generate_metadata
NameError: name 'package' is not defined
```

also, let's bump the version in setup.py, maybe to 0.2.0-dev0

File: py-wacz/wacz/waczindexer.py [hunk 5]
```diff
@@ -196,25 +195,39 @@ def serialize_json_pages(self, pages):
...
-            data['textIndex'] = textIndex
+            package.descriptor['textIndex'] = textIndex
 
-        data['pages'] = [
+        package_dict['pages'] = [
             {'title': page.get('title') or page.get('url'),
              'date': timestamp_to_iso_date(page['timestamp']),
              'url': page['url']} for page in self.pages.values()]
+        
+        return json.dumps(package_dict)
```
Comment (ikreymer) [inline py-wacz/wacz/waczindexer.py:231]: let's add an `indent=2` here to make the json more readable.

**Comment by ikreymer** [comment]
Per https://specs.frictionlessdata.io/data-package/#illustrative-structure, it looks like the `./data` subdirectory is optional.
If so, I think we should not use it then, as the directory structure can be compatible with the pywb collection directory structure, and it would be possible to make the format also compatible with the pywb layout (see: https://pywb.readthedocs.io/en/latest/manual/configuring.html#directory-structure)

**Pull request merged**